### DISCUSSION TITLE
How to integrate `task-list`/`task-item` with `starter-kit`?

### DISCUSSION THREAD

**Discussion by rfgamaral** [ueberdosis/tiptap#2004] [General Questions] [answered]
The official [example](https://tiptap.dev/examples/tasks) for Tasks is a basic editor that pretty much only allows a task list, and that's it. 

I'm trying to integrate the `task-list` and `task-item` extensions with `starter-kit` so that I can have a complete text editor with tasks, but it's not working. Not sure what I'm missing here...

Please take a look at [this CodeSandbox](https://codesandbox.io/s/tiptap-h56o9?file=/src/components/Tiptap.tsx), and look for the line `withTasklistExtension = false` (around line 48). Change that to `true`, and observe the output on the right. The whole thing was turned into a task list.

Thoughts on how to fix this?

**Comment by philippkuehn** [comment, marked as answer]
Removing these lines should fix your issue:

```js
Document.extend({
  content: 'taskList',
}),
```

With this code you define that there can only be one tasklist in your document.

**Comment by rfgamaral** [comment]
@philippkuehn Tried that, but when I type `[ ] ` in the editor, a checkbox does not appear.

**Reply by philippkuehn** [reply]
I can’t reproduce this issue.

**Reply by rfgamaral** [reply]
I've tried it again in the CodeSandbox I posted before (here's the [link](https://codesandbox.io/s/tiptap-h56o9?file=/src/components/Tiptap.tsx) again). I have removed the lines you told me to, and changed `withTasklistExtension = false` to  `true`. Then I tried to add a task with `[ ] `, but as soon as I press "space", the task brackets get removed:

https://user-images.githubusercontent.com/96476/136713447-00b89845-c0ec-4744-bee5-5de494f10f69.mp4

However, after making this video, I changed this:

```ts
extensions.push(
    TaskList,
    TaskItem.extend({
        content: 'inline*',
    }),
)
```

Into this:

```ts
extensions.push(TaskList, TaskItem)
```

And it worked. But I'm confused... What's the purpose of these in the official example?

```ts
const CustomDocument = Document.extend({
  content: 'taskList',
})

const CustomTaskItem = TaskItem.extend({
  content: 'inline*',
})
```

What am I missing?

**Reply by philippkuehn** [reply]
Strange, the `TaskItem` input rule should also work with `content: 'inline*'`. Not sure what’s the issue here right now.

> And it worked. But I'm confused... What's the purpose of these in the official example?

It’s an example to show a todo list document with single line task items. The default task item would allow to contain multiple paragraphs (like the default list item).

**Reply by rfgamaral** [reply]
@philippkuehn

I forked your CodeSandbox template and add the minimal extensions required for Tasks, see [here](https://codesandbox.io/s/tender-microservice-cb5kn?file=/src/App.js). As you can see, this doesn't seem to work:

https://user-images.githubusercontent.com/96476/136922411-5d404fb5-7720-4222-a81e-ec72974d7da2.mp4

If I remove the `content: 'inline*'` from `TaskItem`, it also doesn't work correctly, see [here](https://codesandbox.io/s/lively-river-zrnf8?file=/src/App.js):

https://user-images.githubusercontent.com/96476/136922822-bb02caf9-2da7-422f-afef-baf7a7a2abcc.mp4

**Reply by philippkuehn** [reply]
the first one seems to be an issue, you are right. but the second one works as expected. there are just some missing styles you have to add.

**Reply by rfgamaral** [reply]
I've updated the styles in both examples. The first one still has the same issue, the second one now looks better but exhibits a different issue, not sure if related to the first one:

https://user-images.githubusercontent.com/96476/136942210-12d19c08-5e1d-48bb-a907-50d9d555bf91.mp4

**Reply by philippkuehn** [reply]
I'm not sure what’s the issue here. As I described above it’s expected bahavior that a task item can contain more than a single line of text by default. It’s the same behavior for list items in bullet lists and ordered lists.

**Reply by philippkuehn** [reply]
You can set `content` for list items to `block` or `paragraph` instead.

**Reply by rfgamaral** [reply]
Thank you, @philippkuehn. I'm going to open up a new issue about the one you mentioned it's a bug.

**Comment by rfgamaral** [comment]
@philippkuehn 

> the first one seems to be an issue, you are right. but the second one works as expected. there are just some missing styles you have to add.

Actually, about this, it might not be a bug after all... I mean, I was trying to add a task item to a **paragraph**, which was not supported by `content: 'inline*'`. Removing that made it work, because this is what we have in the TaskItem extension:

```
content() {
    return this.options.nested ? 'paragraph block*' : 'paragraph+'
},
```

Basically, `paragraph` needs to be added to the `content` property. So, in conclusion, I'm now using `nested: true`, which mostly works, here's the updated [demo](https://codesandbox.io/s/tender-microservice-cb5kn?file=/src/App.js), for reference.
